doc/release-notes.md

```diff
@@ -61,6 +61,14 @@ Notable changes
 P2P and network changes
 -----------------------
 
+- This release changes the behavior around processing of unrequested transactions.
+  Previously, a transaction message was always submitted for processing, even if a
+  getdata for this identifier have not been previously issued. Henceforth, such
+  unrequested transaction will be rejected without any processing to mitigate about
+  DoS risks. Bitcoin clients on the network relying on unannounced transaction broadcast
+  should adapt their tx-relay to the compliant tx-request sequence. Peers might be
+  authorized to bypass this check with the PF_RELAY permission. (#20277)
```

Comment:
I don't think light clients can set the flag on the server they connect to. So this will break any such clients that opted to send txs directly. I know that there is no BIP (for obvious reasons) that explains how transaction relay should work, but I'd be cautious breaking a relay mechanism that some software relies on because it has been working in the past. At the very least this breaking change should be communicated to the mailing list.

Reply:
I agree on the mailing list communication. By the way, my concern about breaking tx-broadcast capabilities of such clients was one of the reason for all the back-and-forth on this PR. I expect those clients to be connected to at least few different peers and among them you can assume a subset will be prior versions still supporting processing of unrequested transactions. I would say deployment time give them a window of adaptation.

That said, do you think I should announce on the mailing list now, even before this PR gets merged or only after ?

Reply:
Thanks for taking the suggestion -- I think it would make sense to mention that this is a proposed behavior change for Bitcoin Core, so that if others have feedback on whether this is breaking they can speak up.